externals/kyuubi-trino-engine/src/main/scala/org/apache/kyuubi/engine/trino/schema/SchemaHelper.scala

```
@@ -0,0 +1,100 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kyuubi.engine.trino.schema
+
+import java.util.Collections
+import java.util.Locale
+
+import scala.collection.JavaConverters._
+
+import io.trino.client.ClientStandardTypes._
+import io.trino.client.ClientTypeSignature
+import io.trino.client.Column
+import org.apache.hive.service.rpc.thrift.TCLIServiceConstants
+import org.apache.hive.service.rpc.thrift.TColumnDesc
+import org.apache.hive.service.rpc.thrift.TPrimitiveTypeEntry
+import org.apache.hive.service.rpc.thrift.TTableSchema
+import org.apache.hive.service.rpc.thrift.TTypeDesc
+import org.apache.hive.service.rpc.thrift.TTypeEntry
+import org.apache.hive.service.rpc.thrift.TTypeId
+import org.apache.hive.service.rpc.thrift.TTypeQualifiers
+import org.apache.hive.service.rpc.thrift.TTypeQualifierValue
+
+object SchemaHelper {
+
+  def toTTypeId(typ: ClientTypeSignature): TTypeId = typ.getRawType.toLowerCase(Locale.ROOT) match {
+    case BOOLEAN => TTypeId.BOOLEAN_TYPE
+    case TINYINT => TTypeId.TINYINT_TYPE
+    case SMALLINT => TTypeId.SMALLINT_TYPE
+    case INTEGER => TTypeId.INT_TYPE
+    case BIGINT => TTypeId.BIGINT_TYPE
+    case REAL => TTypeId.FLOAT_TYPE
+    case DOUBLE => TTypeId.DOUBLE_TYPE
+    case DECIMAL => TTypeId.DECIMAL_TYPE
+    case CHAR => TTypeId.CHAR_TYPE
+    case VARCHAR => TTypeId.VARCHAR_TYPE
+    case VARBINARY => TTypeId.BINARY_TYPE
+    case DATE => TTypeId.DATE_TYPE
+    case TIMESTAMP => TTypeId.TIMESTAMP_TYPE
+    case INTERVAL_DAY_TO_SECOND => TTypeId.INTERVAL_DAY_TIME_TYPE
+    case INTERVAL_YEAR_TO_MONTH => TTypeId.INTERVAL_YEAR_MONTH_TYPE
+    case ARRAY => TTypeId.ARRAY_TYPE
+    case MAP => TTypeId.MAP_TYPE
+    case ROW => TTypeId.STRUCT_TYPE
+    case _ => TTypeId.STRING_TYPE
```

shall we raise an error？

---

Done.